identifyConsensus.py

```diff
@@ -10,12 +10,45 @@ def get_parser():
                         nargs="*", help='Paths to HotNet(2) results files.', required=True)
     parser.add_argument('-n', '--networks', nargs="*", required=True,
                         help='List the network that corresponds to each results file.')
+    parser.add_argument('-d', '--delta_choice', action='store_true',
+                        help='Test automated delta selection. Expects a directory of results for each network.')
     parser.add_argument('-o', '--output_file',required=True,
                         help='Output file. Text output by default. Use a .json extension to get JSON output.')
     parser.add_argument('-ms', '--min_cc_size', help='Min CC size.', type=int, default=2)
 
     return parser
-    
+
+# Choose results files to use for consensus (1 per network)
+def choose_results( results_files, networks, min_cc_size, delta_choice ):
+    if delta_choice:
+        networks_results = defaultdict( lambda: set(), dict() )
+        network_dirs = results_files
+        results_files = []
+        for network_dir, network in zip( network_dirs, networks ):
+            results_dirs = [ os.path.join(network_dir, dir)
+                             for dir in os.listdir(network_dir)
+                             if os.path.isdir(os.path.join(network_dir, dir)) & ( dir.split('_')[0] == 'delta' ) ]
+            for results_dir in results_dirs:
+                networks_results[network].add( results_dir + "/results.json" )
+        for network in networks:
+            delta_list = []
+            # Loading results json files and adding entries if k (size) > min_cc_size into a list of
+            # tuples: ( delta, k, p-value, results json file )
+            for results_file in networks_results[network]:
+                with open( results_file ) as f: obj = json.load( f )
+                result_delta_list = [ ( float( obj['parameters']['delta'] ), int( k ), stats['pval'], results_file )
+                                      for k, stats in obj['statistics'].iteritems()
+                                      if int( k ) >= min_cc_size ]
+                delta_list.extend( result_delta_list )
+            # Choosing results json file with the lowest p-value, then the lowest k, then the lowest delta
+            for i in [2, 1, 0]:
```

Comment:
This is nice, but wouldn't it be easier to just create a list of (P-value, k, \delta) tuples, sort them, and then take the first entry?

For example:

    results_files.append( sorted(delta_list)[0][3] )